### PR TITLE
MWPW-120524 Append .html postfix for local links when hosting page ends with .html

### PR DESCRIPTION
**Pull request by chrischrischris**
Resolves: [MWPW-120524](https://jira.corp.adobe.com/browse/MWPW-120524)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/?martech=off
- After: https://c3-htmlpostfix--milo--adobecom.hlx.page/?martech=off

### CODE REVIEW

**Review by rgclayton — COMMENTED** [review]
Are we still waiting for Franklin updates for this?

**Comment by auniverseaway** [comment]
@rgclayton no.

**Comment by auniverseaway** [comment]
This creates some issues with the home page (which you linked to). Can you look into it? I know those blocks are brittle since they're just the internal homepage blocks, but something seems amiss.

**Review by auniverseaway — CHANGES_REQUESTED** [review]
See above.

File: libs/utils/utils.js [hunk 1]
```diff
@@ -427,7 +467,7 @@ function decorateSections(el, isDoc) {
   return [...el.querySelectorAll(selector)].map((section, idx) => {
     const links = decorateLinks(section);
     decorateDefaults(section);
-    const blocks = decorateBlocks(section);
+    const blocks = el.querySelectorAll('div[class]:not(.content)');
```
Comment (auniverseaway) [inline libs/utils/utils.js:470]:
```suggestion
    const blocks = section.querySelectorAll('div[class]:not(.content)');
```

**Review by auniverseaway — APPROVED** [review]

**Comment by rgclayton** [comment]
Seeing a handful of console errors related to rum and lana when pointed to this feature branch that we don't see when pointing to main. Just wanted to verify the errors I'm seeing.

image

**Review by rgclayton — APPROVED** [review]
All console errors are gone after that last update. Thanks Chris!

**Pull request merged**